### PR TITLE
fix: function:serve shows undefined port in url

### LINKED ISSUE #4116: netlify functions:serve shows an undefined port in loaded function URL
**Describe the bug**

When using `netlify functions:serve` the URLs for the loaded functions have an `undefined` port.

```
◈ Loaded function test-function (​http://localhost:undefined/.netlify/functions/test-function​).
```

Directly after the list of loaded functions, the default port is listed.

```
◈ Functions server is listening on 9999
```

If you copy the URL for the function and replace `undefined` with `9999` the function works and returns its response.

**To Reproduce**

1. Create a function in `netlify/functions`
2. Run `netlify functions:serve`
3. Observe the loaded function URL has the port `undefined`

**Configuration**

```
  System:
    OS: Linux 4.19 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (12) x64 Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz
    Memory: 23.45 GB / 50.14 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 16.7.0 - ~/.nvm/versions/node/v16.7.0/bin/node
    Yarn: 1.16.0 - /mnt/c/Users/Ian/AppData/Roaming/npm/yarn
    npm: 7.20.3 - ~/.nvm/versions/node/v16.7.0/bin/npm
    Watchman: 2021.06.07.00 - /home/linuxbrew/.linuxbrew/bin/watchman
  npmGlobalPackages:
    netlify-cli: 8.13.0
```

**Expected behavior**

The URL would contain port `9999` instead of `undefined`.

### PR DESCRIPTION
**Pull request by karagulamos**
#### Summary

Fixes #4116 

image

When a Netlify site is served using `netlify functions:serve` locally, the function URL shown at the terminal contains an `undefined` port which is incorrect albeit confusing to users when they click on it. This is caused by the use of an undefined [`NetlifyFunction:settings:port`](https://github.com/netlify/cli/blob/main/src/lib/functions/netlify-function.js#L117) property composed within the [`NetlifyFunction:url`](https://github.com/netlify/cli/blob/main/src/lib/functions/netlify-function.js#L117) (getter) property. This PR fixes the issue and includes relevant tests.

**NOTE**: Although I scoured the codebase for a valid usage of `NetlifyFunction:settings:port` to no avail, I’m not particularly fond of the current solution but left it there with comments (and for compatibility) should anyone wish to take a closer look.

image

### CODE REVIEW

File: src/lib/functions/netlify-function.js [hunk 1]
```diff
@@ -114,7 +114,12 @@ class NetlifyFunction {
   }
 
   get url() {
-    return `http://localhost:${this.settings.port}/.netlify/functions/${this.name}`
+    // This line fixes the issue here https://github.com/netlify/cli/issues/4116
+    // Not sure why `settings.port` was used here nor does a valid reference exist.
+    // However, it remains here to serve whatever purpose for which it was added.
+    const port = this.settings.port || this.settings.functionsPort
+
+    return `http://localhost:${port}/.netlify/functions/${this.name}`
```
Comment (lukasholzer) [inline src/lib/functions/netlify-function.js:122]: Maybe you want to construct the url based on the `URL`.

```
new URL({
  port,
  ...
}).href
```

Comment (lukasholzer) [reply src/lib/functions/netlify-function.js:122]: @erezrokah are the functions always served on the host `localhost` or could it be `0.0.0.0` as well?

**Review by lukasholzer — DISMISSED** [review]

**Comment by lukasholzer** [comment]
Thx for adapting the tests! 🐳

**Comment by erezrokah** [comment]
@karagulamos, do you mind resolving the merge conflict? The test file was renamed in https://github.com/netlify/cli/pull/4164

**Review by erezrokah — APPROVED** [review]
Going to approve per @lukasholzer review so we can merge this

**Pull request merged**